File: android/src/main/kotlin/br/com/keyboard_utils/manager/KeyboardOptions.kt
package br.com.keyboard_utils.manager

import org.json.JSONObject

/**
 * Created by Wilson Martins on 2019-10-25.
 */
class KeyboardOptions(private val isKeyboardOpen: Boolean, private val height: Int) {
    fun toJson(): String = JSONObject().apply {
        put("isKeyboardOpen", isKeyboardOpen)
        put("keyboardHeight", height)
    }.toString()
}

File: lib/keyboard_options.dart
class KeyboardOptions {
  KeyboardOptions({required this.isKeyboardOpen, required this.keyboardHeight});

  final bool isKeyboardOpen;
  final double keyboardHeight;

  KeyboardOptions.fromJson(Map<String, dynamic> json)
      : isKeyboardOpen = json['isKeyboardOpen'],
        keyboardHeight = json['keyboardHeight'].toDouble();

  Map<String, dynamic> toJson() {
    return {
      'isKeyboardOpen': isKeyboardOpen,
      'keyboardHeight': keyboardHeight,
    };
  }
}
